vault/token_store.go

```diff
@@ -783,12 +809,13 @@ func (ts *TokenStore) create(ctx context.Context, entry *TokenEntry) error {
 		entry.ID = entryUUID
 	}
 
-	saltedID, err := ts.SaltID(ctx, entry.ID)
+	entry.Version = 2
+
+	te, err := ts.lookupTokenNonLocked(ctx, entry.ID, true)
 	if err != nil {
 		return err
 	}
-	exist, _ := ts.lookupSalted(ctx, saltedID, true)
-	if exist != nil {
+	if te != nil {
```

Comment:
I think to ensure that the ID isn't duplicate we also need to check that a salted version doesn't exist. Sad but true.

Reply:
Its already covered. The function `lookupTokenNonLocked` takes in a plaintext token and tries to lookup using both idHMAC and saltedID, if the former fails.

Reply:
Ah yeah, fair.